### PR TITLE
Chunk all types

### PR DESCRIPTION
**Pull request by gareth-j**
* **Summary of changes** (Bug fix, feature, docs update, ...)

Adds chunk argument to all standardise functions and storage_type.read_file functions.

### CODE REVIEW

File: openghg/standardise/_standardise.py [hunk 1]
```diff
@@ -119,6 +120,9 @@ def standardise_surface(
             See https://zarr.readthedocs.io/en/stable/api/codecs.html for more information on compressors.
         filters: Filters to apply to the data on storage, this defaults to no filtering. See
             https://zarr.readthedocs.io/en/stable/tutorial.html#filters for more information on picking filters.
+        chunks: Chunk schema to use when storing data the NetCDF. It expects a dictionary of dimension name and chunk size,
+            for example {"time": 100}. If None then a chunking schema will be set automatically by OpenGHG as per the <link to documentation>.
```
Comment (rt17603) [inline openghg/standardise/_standardise.py:124]: Would it be worth adding explicit TODO lines within all the doc-strings to make sure we remember to add details for "link to documentation" before we finish off all the changes?

File: openghg/store/_boundary_conditions.py [hunk 2]
```diff
@@ -96,6 +97,9 @@ def read_file(
                 See https://zarr.readthedocs.io/en/stable/api/codecs.html for more information on compressors.
             filters: Filters to apply to the data on storage, this defaults to no filtering. See
                 https://zarr.readthedocs.io/en/stable/tutorial.html#filters for more information on picking filters.
+            chunks: Chunk schema to use when storing data the NetCDF. It expects a dictionary of dimension name and chunk size,
+                for example {"time": 100}. If None the a chunking schema will be set automatically by OpenGHG as per the <link to documentation>.
```
Comment (rt17603) [inline openghg/store/_boundary_conditions.py:101]:
```suggestion
                for example {"time": 100}. If None then a chunking schema will be set automatically by OpenGHG as per the <link to documentation>.
```

File: openghg/store/_boundary_conditions.py [hunk 3]
```diff
@@ -139,7 +143,10 @@ def read_file(
 
         filepath = next(iter(unseen_hashes.values()))
 
-        with open_dataset(filepath) as bc_data:
+        if chunks is None:
```
Comment (rt17603) [inline openghg/store/_boundary_conditions.py:146]: Just to check, is this a placeholder to use for implementing the default chunking scheme in another PR? Or have I misinterpreted that?

File: openghg/store/_eulerian_model.py [hunk 4]
```diff
@@ -68,6 +69,9 @@ def read_file(
                 See https://zarr.readthedocs.io/en/stable/api/codecs.html for more information on compressors.
             filters: Filters to apply to the data on storage, this defaults to no filtering. See
                 https://zarr.readthedocs.io/en/stable/tutorial.html#filters for more information on picking filters.
+            chunks: Chunk schema to use when storing data the NetCDF. It expects a dictionary of dimension name and chunk size,
+                for example {"time": 100}. If None the a chunking schema will be set automatically by OpenGHG as per the <link to documentation>.
```
Comment (rt17603) [inline openghg/store/_eulerian_model.py:73]:
```suggestion
                for example {"time": 100}. If None then a chunking schema will be set automatically by OpenGHG as per the <link to documentation>.
```

File: openghg/store/_obscolumn.py [hunk 5]
```diff
@@ -79,6 +80,9 @@ def read_file(
                 See https://zarr.readthedocs.io/en/stable/api/codecs.html for more information on compressors.
             filters: Filters to apply to the data on storage, this defaults to no filtering. See
                 https://zarr.readthedocs.io/en/stable/tutorial.html#filters for more information on picking filters.
+            chunks: Chunk schema to use when storing data the NetCDF. It expects a dictionary of dimension name and chunk size,
+                for example {"time": 100}. If None the a chunking schema will be set automatically by OpenGHG as per the <link to documentation>.
```
Comment (rt17603) [inline openghg/store/_obscolumn.py:84]:
```suggestion
                for example {"time": 100}. If None then a chunking schema will be set automatically by OpenGHG as per the <link to documentation>.
```

**Review by rt17603 — APPROVED** [review]
Looks good - a few small typos to change and a few questions but don't need to re-review this so have added an approval now.

File: openghg/standardise/_standardise.py (hunk 1 above)
Comment (gareth-j) [reply openghg/standardise/_standardise.py]: Done

File: openghg/store/_boundary_conditions.py (hunk 2 above)
Comment (gareth-j) [reply openghg/store/_boundary_conditions.py]: Fixed.

**Pull request merged**